### ISSUE TITLE
[Automation] Automate ArgoCD onboarding and additional features

### ISSUE THREAD

**Issue by HumairAK** [operate-first/SRE#305]
Let's automate the entire argocd onboarding procedure for the `apps` repository, we can leave worded instructions for argocd-apps repo for now. 

The result should be a tool that can automate the steps/changes made to apps repo as described in the onboarding docs [here](https://github.com/operate-first/support/blob/main/docs/onboarding_to_argocd.md). 

The instructions in the docs should also be updated to reflect the usage of this tool.

**Comment by HumairAK** [comment]
Let's wait on a resolution for operate-first/SRE#299 to be resolved before proceeding with this.

**Comment by HumairAK** [comment]
I actually think we may include the updates to `argocd-app` as part of this (and thus the automate the full onboarding process), by just requiring the invoker to provide a path to the argocd-apps cloned repo. wdyt?

**Comment by tumido** [comment]
@HumairAK previous comment made into a separate issue operate-first/SRE#309
